libhoney/src/main/java/io/honeycomb/libhoney/Options.java

```diff
@@ -75,7 +79,13 @@ public String getWriteKey() {
      * @see Builder#setDataset(String)
      */
     public String getDataset() {
-        return dataset;
+        if (isClassic()) {
+            return dataset;
+        }
+        if (dataset == null || dataset.trim().length() == 0) {
+            return DEFAULT_NON_CLASSIC_DATASET;
+        }
+        return dataset.trim();
```

Comment:
we also talked about warning users when the trimming changes the value, so they can fix the discrepancy (like we did in the [go beeline](https://github.com/honeycombio/beeline-go/blob/dcd5cac17320d77d7242d157e4dea6c757edbd03/beeline.go#L146))